compiler/rustc_resolve/src/late.rs

```diff
@@ -540,6 +586,9 @@ struct LateResolutionVisitor<'a, 'b, 'ast> {
     /// The current set of local scopes for lifetimes.
     lifetime_ribs: Vec<LifetimeRib>,
 
+    /// We are looking for lifetimes in an elision context.
+    lifetime_elision_candidates: Option<FxIndexMap<LifetimeRes, LifetimeElisionCandidate>>,
```

Comment:
This is also a diagnostics stuff?
It can be moved to `diagnostic_metadata` then.

Reply:
Ok, looks like I misinterpreted.
The `Set<LifetimeRes>` part is significant for the actual elision logic, only `LifetimeElisionCandidate` is for diagnostics.